NIO tests no longer build for 32-bit Android armv7 with the latest 6.1 March 20 snapshot

### Expected behavior
The tests to cross-compile fine, as they did [yesterday with the prior 6.1 March 12 snapshot, though strangely this now failing file is not listed for the armv7 build](https://github.com/finagolfin/swift-android-sdk/actions/runs/13985105885/job/39157355759#step:19:1953).

### Actual behavior
A new [compilation error for two tests](https://github.com/finagolfin/swift-android-sdk/actions/runs/14005493499/job/39218841508#step:19:1766):
```
Tests/NIOPosixTests/NonBlockingFileIOTest.swift:326:47: error: arithmetic operation '2147483647 + 10' (on type 'Int') results in an overflow
 324 |                 let content = try self.fileIO.read(
 325 |                     fileHandle: filehandle,
 326 |                     byteCount: Int(Int32.max) + 10,
     |                                               `- error: arithmetic operation '2147483647 + 10' (on type 'Int') results in an overflow
 327 |                     allocator: .init(),
 328 |                     eventLoop: self.eventLoop

Tests/NIOPosixTests/NonBlockingFileIOTest.swift:1407:47: error: arithmetic operation '2147483647 + 10' (on type 'Int') results in an overflow
1405 |                 let content = try await self.fileIO.read(
1406 |                     fileHandle: filehandle,
1407 |                     byteCount: Int(Int32.max) + 10,
     |                                               `- error: arithmetic operation '2147483647 + 10' (on type 'Int') results in an overflow
1408 |                     allocator: .init()
1409 |                 )
```

### Steps to reproduce

Download my [CI-generated Android SDK bundle for 6.1](https://github.com/finagolfin/swift-android-sdk/actions/runs/14005493499/attempts/1) and build for armv7 using [the instructions there](https://github.com/finagolfin/swift-android-sdk?tab=readme-ov-file#cross-compiling-and-testing-swift-packages-with-the-android-sdk-bundle).

### SwiftNIO version/commit hash

771810f

### System & version information

Swift 6.1 cross-compiling from Ubuntu 24.04 x86_64 and macOS 13 x86_64

May be related to swiftlang/swift#79707, which I notice is in the latest 6.1 tag but hasn't made it into a trunk tag yet, which would explain why trunk still compiles fine. This seems like a legit bug that the compiler should have flagged on armv7 before though, so I'm filing it here.

--- a/Tests/NIOPosixTests/NonBlockingFileIOTest.swift
+++ b/Tests/NIOPosixTests/NonBlockingFileIOTest.swift
@@ -323,7 +323,9 @@ class NonBlockingFileIOTest: XCTestCase {
             { (filehandle, path) -> Void in
                 let content = try self.fileIO.read(
                     fileHandle: filehandle,
-                    byteCount: Int(Int32.max) + 10,
+                    // There's a runtime check above, use overflow addition to stop the compilation
+                    // from failing on 32-bit platforms.
+                    byteCount: Int(Int32.max) &+ 10,
                     allocator: .init(),
                     eventLoop: self.eventLoop
                 ).wait()
@@ -1404,7 +1406,9 @@ extension NonBlockingFileIOTest {
             { (filehandle, path) -> Void in
                 let content = try await self.fileIO.read(
                     fileHandle: filehandle,
-                    byteCount: Int(Int32.max) + 10,
+                    // There's a runtime check above, use overflow addition to stop the compilation
+                    // from failing on 32-bit platforms.
+                    byteCount: Int(Int32.max) &+ 10,
                     allocator: .init()
                 )
                 XCTAssertEqual(String(buffer: content), "some-dummy-content")

